File: spring/src/main/java/com/example/shoppingCart/models/Orde_r.java
package com.example.shoppingCart.models;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Entity
@Data
public class Orde_r {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime orderDate;

    @OneToMany(mappedBy = "order", cascade = CascadeType.PERSIST)
    private List<OrderDetail> orderDetails;

    @ManyToOne
    private Customer customer;

    private UUID txnId;


}

File: angular/shopping-cart/src/app/models/Order.ts
import { IOrderDetail } from './OrderDetail';
import { ICustomer } from './Customer';

export interface IOrder {
    id: number;
    orderDate: Date;
    orderDetails: IOrderDetail[];
    customer: ICustomer;
    txnId: any;
}

export class Order implements IOrder {
    id: number = 0;
    orderDate: Date;
    orderDetails: IOrderDetail[];
    customer: ICustomer;
    txnId: any;
}